Cargo.toml

```diff
@@ -76,12 +73,23 @@ members = ["libs/test-utils"]
 exclude = ["scripts/attestation_agent/app"]
 
 [features]
-default = ["snapshot-overlayfs", "signature-cosign", "keywrap-grpc"]
-kata-cc = ["encryption", "keywrap-ttrpc", "snapshot-overlayfs", "signature-cosign", "signature-simple", "getresource"]
-kata-cc-s390x = ["encryption", "keywrap-ttrpc", "snapshot-overlayfs", "signature-simple", "getresource"]
-enclave-cc = ["encryption", "keywrap-native", "eaa-kbc", "snapshot-unionfs", "signature-simple", "getresource", "signature-cosign"]
+default = ["snapshot-overlayfs", "signature-cosign-rustls", "keywrap-grpc", "rustls-tls"]
+
+# This will be based on `ring` dependency
+kata-cc-rustls = ["encryption-ring", "keywrap-ttrpc", "snapshot-overlayfs", "signature-cosign-rustls", "signature-simple", "getresource", "rustls-tls"]
+enclave-cc-rustls = ["encryption-ring", "keywrap-native", "eaa-kbc", "snapshot-unionfs", "signature-simple", "getresource", "signature-cosign-rustls", "rustls-tls"]
+
+# This will be based on `openssl` dependency
+kata-cc-native = ["encryption-openssl", "keywrap-ttrpc", "snapshot-overlayfs", "signature-cosign-native", "signature-simple", "getresource", "native-tls"]
+enclave-cc-native = ["encryption-openssl", "keywrap-native", "eaa-kbc", "snapshot-unionfs", "signature-simple", "getresource", "signature-cosign-native", "native-tls"]
+
+native-tls = ["oci-distribution/native-tls"]
+rustls-tls = ["oci-distribution/rustls-tls"]
```

Comment:
can we rename `native-tls` and `rustls-tls` with prefix since these two features seems global and may confusing.